src/App.tsx

```diff
@@ -1,17 +1,150 @@
-import React from 'react';
-import './App.css';
+import React, { useState, useMemo } from 'react';
+import './App.scss';
+import { Todos } from './types/Todos';
+// import { TodoList } from './components/TodoList/TodoList';
+import { TodoInfo } from './components/TodoInfo/TodoInfo';
 
 import users from './api/users';
+import todos from './api/todos';
 
 const App: React.FC = () => {
+  const [title, setTitle] = useState('');
+  const [user, setUser] = useState('');
+  const [todo, setTodo] = useState(todos);
+  const [status, setStatus] = useState('not completed');
+  const [errorUser, setErrorUser] = useState<string | null>(null);
+  const [errorTitle, setErrorTitle] = useState<string | null>(null);
+
+  const hangleTitle = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const newTitle = event.target.value;
+
+    setTitle(newTitle.replace(/[^a-zA-Z0-9А-Яа-я\s]/g, ''));
+    setErrorTitle(null);
+  };
+
+  const validate = () => {
+    if (!user) {
+      setErrorUser('Please choose a user');
+    }
+
+    if (!title.trim()) {
+      setErrorTitle('Please enter the title');
+    }
+
+    if (!user || !title.trim()) {
+      return false;
+    }
+
+    return true;
+  };
+
+  const addTodo = (event: React.SyntheticEvent) => {
+    event.preventDefault();
+    if (validate()) {
+      const searchingUser = users.find(({ name }) => name === user);
+
+      const newTodo = {
+        id: todo[todo.length - 1].id + 1,
+        title,
+        userId: searchingUser ? searchingUser.id : 0,
+        completed: status === 'completed',
+      };
+
+      setTodo((current) => [...current, newTodo]);
+      setTitle('');
+      setUser('');
+      setStatus('not completed');
+    }
+  };
+
+  const newArr = (arrTodo: Omit<Todos, 'user'>[]) => arrTodo.map((oneTodo) => ({
+    ...oneTodo,
+    user: users.find((oneUser) => oneUser.id === oneTodo.userId),
+  }));
+
+  const preparedTodos: Todos[] = useMemo(() => newArr(todo), [todo]);
```

Comment:
```suggestion
  const preparedTodos: Todos[] = useMemo((todos) => todos.map((todo) => ({
    ...todo,
    user: users.find((oneUser) => oneUser.id === todo.userId),
  }), [todos]);
```

Or rename newArr - non-understandable name)